imgui/src/lib.rs

```diff
@@ -115,26 +116,44 @@ impl Context {
 
 /// A temporary reference for building the user interface for one frame
 #[derive(Debug)]
-pub struct Ui<'ui> {
-    ctx: &'ui Context,
-    font_atlas: Option<std::cell::RefMut<'ui, SharedFontAtlas>>,
-    // imgui isn't mutli-threaded -- so no one will ever access twice.
-    buffer: std::cell::UnsafeCell<string::UiBuffer>,
-}
+pub struct Ui(pub(crate) cell::UnsafeCell<()>);
+
+/// This is our internal buffer that we use for the Ui object.
+///
+/// We edit this buffer
+static mut BUFFER: cell::UnsafeCell<string::UiBuffer> =
```

Comment:
Almost certainly no point in both `static mut` and `UnsafeCell` here.

Also, I don't think your arguments around simplicity really make sense that much here. I think the important part of the `Ui<'ui>` to remove was just the lifetime. This being a ZST doesn't really seem to help anything, and it does make adding any additional state `Ui` much harder, it has to be global variables.